### DISCUSSION TITLE
Convert typing.Union to new PEP 604 syntax (using ' | ')

### DISCUSSION THREAD

**Discussion by kumaraditya303** [microsoft/pylance-release#2652] [Enhancement]
### Discussed in https://github.com/microsoft/pyright/discussions/2550

<div type='discussions-op-text'>

<sup>Originally posted by **kumaraditya303** November  9, 2021</sup>
Feature request to convert legacy typing.Union syntax to newer PEP 604 syntax for large codebases where manually converting will take a lot of effort. Thanks </div>

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
